src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExplicitConversion.cs

```diff
@@ -132,11 +132,14 @@ public bool Bind()
 
                 // But it failed, and so we know that the constant is not in range
 
+                Debug.Assert(
+                    _typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ErrorType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ArgumentListType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ParameterModifierType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_NullableType);
```

Comment:
In general, this style makes me nervous because it means if I were to add a value to TypeKind, whereas the old approach would have flagged it with an assert, this way won't, since it's now relying on me manually have written out the complete set of values between those I'm negatively and positively checking for.